src/api/middleware/user-data.middleware.js

```diff
@@ -0,0 +1,20 @@
+import {Exception} from "../exceptions/Exception.js";
+
+function validateStringNotEmpty(string, errorMessagePrefix, errors) {
+    if (!string || string.trim().length === 0) {
+        errors.push(`${errorMessagePrefix} is empty`);
+    }
+}
+
+export const checkUserDataMiddleware = (req, res, next) => {
+    const userData = req.body;
+    const errors = [];
+    validateStringNotEmpty(userData.id, "User id", errors);
+    validateStringNotEmpty(userData.name, "User name", errors);
```

Comment:
I've implemented it as you advised before.
In case of throwing an exception in the validator, we don't collect all possible errors, we get only the first one.  
In this implementation, I tried to follow an approach that is implemented in ajv (validator returns errors). 

If I'm mistaken just let me know, please, and I'll improve it.